### ISSUE TITLE
Does a pixel have multiple color in CV?

### ISSUE THREAD

**Issue by thsno02** [thsno02/ocr#9]
- [x] what pixels are?
- [x] does a pixel have many color set?

**Comment by thsno02** [comment]
# The Definition of Pixels

[merriam-webste](https://www.merriam-webster.com/dictionary/pixel): 
1. any of the small discrete elements that together constitute an image (as on a television or digital screen)
2. any of the detecting elements of a solid-state optical sensor (such as a CCD or CMOS device)

The pixels in CV refer to the first definition, which are minute areas of illumination an image in the screen. Informally, pixels are servered as the color ~dots~ block.

[Wikipedia](https://en.wikipedia.org/wiki/Pixel):
>In [digital imaging](https://en.wikipedia.org/wiki/Digital_imaging), a pixel (abbreviated px), pel,[[1]](https://en.wikipedia.org/wiki/Pixel#cite_note-1) or picture element[[2]](https://en.wikipedia.org/wiki/Pixel#cite_note-2) is the smallest addressable element in a [raster image](https://en.wikipedia.org/wiki/Raster_graphics), or the smallest addressable element in an [all points addressable](https://en.wikipedia.org/wiki/All_points_addressable) [display device](https://en.wikipedia.org/wiki/Display_device); so it is the smallest controllable element of a picture represented on the screen.

>In [computer graphics](https://en.wikipedia.org/wiki/Computer_graphics) and [digital photography](https://en.wikipedia.org/wiki/Digital_photography), a raster graphic represents a two-dimensional picture as a rectangular matrix or grid of square [pixels](https://en.wikipedia.org/wiki/Pixel), viewable via a [computer display](https://en.wikipedia.org/wiki/Computer_display), [paper](https://en.wikipedia.org/wiki/Paper), or other display medium.<sup>[[1]](#ft1)</sup>

>All points addressable (APA), or pixel addressable, in the context of a dot matrix on a [computer monitor](https://en.wikipedia.org/wiki/Computer_monitor) or any [display device](https://en.wikipedia.org/wiki/Display_device) consisting of a [pixel](https://en.wikipedia.org/wiki/Pixel) array, refers to an arrangement whereby bits or cells can be individually manipulated, as opposed to rewriting the whole array, or regions such as characters, every time a change is needed.<sup>[[2]](#ft2)</sup>

Formally, a pixel is the minimum addressable element in a raster image which is a two-diemsional picture viewed as a rectangular matrix of grid of square pixels.

>Each pixel is a [sample](https://en.wikipedia.org/wiki/Sampling_(signal_processing)) of an original image; more samples typically provide more accurate representations of the original. The [intensity](https://en.wikipedia.org/wiki/Intensity_(physics)) of each pixel is variable. In color imaging systems, a color is typically represented by three or four component intensities such as [red, green, and blue](https://en.wikipedia.org/wiki/RGB_color_model), or [cyan, magenta, yellow, and black](https://en.wikipedia.org/wiki/CMYK_color_model).<sup>[[3]](#ft3)</sup>

## The Representation

Each pixel corresponds to a smaple of an orignal image, and presents in one, three or four component intensities such as grayscale or RGB or CMYK. In a grayscale system, the value of each pixel ranges from 0 to 255, the value approaching 0 means the color is close to black while 1 means to white. In RGB system, each pixel is represented as a three dimension vector, `(red, green, blue)`, and each of them ranges from 0 to 255. In CMYK, a four-dimensional vector,  `(cyan, magenta, yellow, and black)`, represents each pixel, and the range is same as the previous. In a nutshell, 255 means the actual color of white, red, green, blue, and etc, and vice versa.

Meanwhile, in computer, the picture is not stored as color set, but as number ranging from 0 to 255, 8 bit data. Formally, a picture is represented as a matrix, each element in the matrix is a vector if it is a colorful picture.

## The Resolution

> Image resolution is typically described in PPI, which refers to how many pixels are displayed per inch of an image.  Images with lower resolutions have fewer pixels, and if those few pixels are too large (usually when an image is stretched), they can become visible like the image below. <sup>[[4]](#ft4)</sup>

In other words, the resolution is measured by PPI, which is the number of pixels per inch of an image. 

Resizing a picture:

+ downsize: maybe average the values of nearby pixels
+ upsize: guess the best color set

# Footnotes

<a name="ft1">[1]</a>: https://en.wikipedia.org/wiki/Raster_graphics

<a name="ft2">[2]</a>: https://en.wikipedia.org/wiki/Dot_matrix#All_points_addressable

<a name="ft3">[3]</a>: https://en.wikipedia.org/wiki/Pixel

<a name="ft4">[4]</a>: https://guides.lib.umich.edu/c.php?g=282942&p=1885350
